pkg/playground/react-demo-terminal.jsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "react",
+    "base1/term",
+], function(React, Term) {
+
+"use strict";
+
+/*
+ * A terminal component that communicates over a cockpit channel.
+ *
+ */
+var Terminal = React.createClass({
+  propTypes: {
+    cols: React.PropTypes.number,
+    rows: React.PropTypes.number,
+    channel: React.PropTypes.object.isRequired,
+    onTitleChanged: React.PropTypes.func
+  },
+
+  getDefaultProps: function () {
+    return {
+      cols: 80,
+      rows: 25
+    };
+  },
+
+  componentWillMount: function () {
```

Comment:
React caches the return value of `getInitialState` between components. `componentWillMount` is always called before mounting it into the DOM. I also think that it's nicely symmetric with `componentDidMount`, in which I hook the term.js widget into the DOM.